build/bin/remove_unneeded

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Sanity check: the file is created by the Dockerfile.
+if [ ! -f /in-the-container ]; then
+  echo "Don't run this outside the container!"
+  exit 1
+fi
+
+# Array of basic tools needed by entrypoint and user_setup.
+basicNeed=("bin/sh" "/bin/bash" "/usr/bin/coreutils")
+libraries=()
+
+# Get all that is needed by basicNeed
```

Comment:
this comment is a bit minimal - you're filtering on shared libraries, identified by `\.so`.